src/crystal/compiler_rt.cr

```diff
@@ -0,0 +1,38 @@
+{% skip_file unless flag?(:compiler_rt) %}
+
+fun __mulodi4(a : Int64, b : Int64, overflow : Int32*) : Int64
+  n = 64
+  min = Int64::MIN
+  max = Int64::MAX
+  overflow.value = 0
+  result = a &* b
+  if a == min
+    if b != 0 && b != 1
+      overflow.value = 1
+    end
+    return result
+  end
+  if b == min
+    if a != 0 && a != 1
+      overflow.value = 1
+    end
+    return result
+  end
+  sa = a >> (n - 1)
```

Comment:
Is it OK to use `-` here instead of `-&`? (same everywhere in this function)

Reply:
Good catch. Fixed in the last commit. I will squash that after approval.